custom/inddex/sqldata.py

```diff
@@ -60,9 +60,8 @@ class FoodCodeData(FoodCodeDataSource):
 
     @property
     def rows(self):
-        food_codes = [int(x['food_code']) for x in self.get_data() if x['food_code'] is not None]
-
-        return sorted(set(food_codes))
+        food_codes = {int(x['food_code']) for x in self.get_data() if x['food_code']}
```

Comment:
Is it safe to assume that `x['food_code']` is never 0?

Reply:
It's a string type, so if anything, it'd probably be `"0"`, which is truthy.

Looking at it, I'm actually not sure why this coerces to an `int`.  The codes _are_ numeric, but that doesn't make them integers.  Indeed, I believe the only place they're used, they're promptly converted back to strings:
https://github.com/dimagi/commcare-hq/blob/17365b2031ab56d9809819124489995340ca4bdb/custom/inddex/filters.py#L152-L160

Must be for sorting?  Perhaps something like this would be better?

```python
    @property
    def rows(self):
        def is_numeric(code):
            return isinstance(code, str) and code.isnumeric()
        food_codes = filter(is_numeric, self.get_data())
        return sorted(set(food_codes), key=int)
```

Reply:
Gonna merge this PR as-is for now and possibly address this later

Reply:
Yeah. I was wondering whether `FoodCodeData.rows` or `FoodCodeFilter.food_codes` were used for creating buckets, in which case it might make sense to build a sorted set where decimal values are converted to integers. But it looks like they are used to populate filter options.

If all food codes are definitely integers or blank, then I think this is OK as it is. Converting to int and then straight back to string isn't great, but it's fine.

But if it is possible for food codes to be decimals, we should use `key=float` or `key=Decimal`, otherwise we could end up with non-integer values returned in the wrong order:

```python
    >>> food_codes = [3, 2.9, 2.2, 1, 1]
    >>> sorted(set(food_codes), key=int)
    [1, 2.9, 2.2, 3]
```